lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -589,24 +589,77 @@ migrateManually tr proxy defaultFieldValues =
-            >> Sqlite.columns fee_policyInfo
-        Sqlite.finalize fee_policyInfo
-        case row of
-            [PersistText t] -> do
-                (a:b:_) <- pure (T.splitOn " + " t)
-                let newVal = a <> " + " <> b
+        let field = (DBField ProtocolParametersFeePolicy)
+            defaultValue = "0 + 0x"
+
+        isFieldPresent conn field >>= \case
+            TableMissing ->
+                traceWith tr $ MsgManualMigrationNotNeeded field
+            ColumnMissing -> do
+                traceWith tr $ MsgManualMigrationNeeded field defaultValue
                 query <- Sqlite.prepare conn $ T.unwords
-                    [ "UPDATE protocol_parameters"
-                    , "SET fee_policy", "=", newVal
+                    [ "ALTER TABLE", tableName field
+                    , "ADD COLUMN", fieldName field
+                    , fieldType field, "NOT NULL"
+                    , "DEFAULT", defaultValue
                     , ";"
                     ]
-                Sqlite.step query *> Sqlite.finalize query
-            _ -> pure ()
+                _ <- Sqlite.step query
+                Sqlite.finalize query
+            ColumnPresent -> do
+                fee_policyInfo <- Sqlite.prepare conn $ T.unwords
+                    [ "SELECT"
+                    , fieldName field
+                    , "FROM"
+                    , tableName field
+                    , ";"
+                    ]
+                row <- Sqlite.step fee_policyInfo
+                    >> Sqlite.columns fee_policyInfo
+                Sqlite.finalize fee_policyInfo
+                case row of
+                    [PersistText t] -> do
+                        traceWith tr $ MsgManualMigrationNeeded field t
+                        [a, b, c] <- pure (T.splitOn " + " t)
+
+                        -- update fee policy
+                        let newVal = a <> " + " <> b
+                        query <- Sqlite.prepare conn $ T.unwords
+                            [ "UPDATE"
+                            , tableName field
+                            , "SET"
+                            , fieldName field
+                            , "= '" <> newVal <> "'"
+                            , ";"
+                            ]
+                        Sqlite.step query *> Sqlite.finalize query
+
+                        -- update stake key val
+                        updateStakeKeyVal (DBField ProtocolParametersKeyDeposit) c
+
+                    _ -> fail "Couldn't query fee value!"
+      where
+        updateStakeKeyVal :: DBField -> Text -> IO ()
+        updateStakeKeyVal stakeField c = do
+            isFieldPresent conn stakeField >>= \case
+                TableMissing ->
+                    -- this invariant must hold due to the parent context
+                    fail "ProtocolParameters table missing"
+                ColumnMissing -> do
+                    traceWith tr $ MsgManualMigrationNotNeeded stakeField
```

Comment:
Questionable, but well...

Reply:
Actually, a migration is needed. And that migration is `addKeyDepositIfMissing`. Which is a bit unfortunate, because now, if we compose the two actions above (`addKeyDepositIfMissing` and `updateFeeValue`), we end up in a situation where the column is created with a possibly wrong value...

I'd suggest adding the `key_deposit` column here, instead of `addKeyDepositIfMissing`.